Add Worker Limits to AWS Resource Reference + Improve Worker Docs (#637)
astronomer/docs: astro/configure-deployment.md

:::

## Configure Deployment Resources

Once you create a Deployment, you can choose to modify its resources and configurations to best fit the needs of your tasks. Specifically, you can modify these two components:

- Worker(s)
- Scheduler

For reference, Astro supports the `AU` as the primary resource unit. In this context,

- 1 AU = 0.1 CPU, .375 GB Memory
- 10 AU = 1 CPU, 3.75 GB Memory

Read below for guidelines on how to configure each component.

### Worker Resources

> A worker is responsible for executing tasks, which are first scheduled and queued by the Scheduler. On Astro, task execution is powered by the [Celery Executor](https://airflow.apache.org/docs/apache-airflow/stable/executor/celery.html) with [KEDA](https://www.astronomer.io/blog/the-keda-autoscaler), which enables workers to auto-scale between 1 and 10 depending on real-time workload. Each worker is a Kubernetes Pod that is hosted within a Kubernetes Node in your Astro Cluster.

Reviewer: What limits the workers at 10?  Isn't there some sort of configuration that allows for more than 10 workers?

Author: It actually is a hard cap at 10 right now: https://github.com/astronomer/airflow-operator/blob/fb8989d99e66047c0d71393f0b0b2341494e170c/config/crd/bases/airflow.apache.org_workers.yaml#L165

Actively working on a new "Min/Max Worker Count" slider though: https://github.com/astronomer/harmony/issues/95

Reviewer 2: Out of scope for this PR, but I feel this is another knob that we could take away from the customer per our conversation yesterday. We already have a control for capping node count.

> A worker is responsible for executing tasks, which are first scheduled and queued by the Scheduler. On Astro, task execution is powered by the [Celery Executor](https://airflow.apache.org/docs/apache-airflow/stable/executor/celery.html) with [KEDA](https://www.astronomer.io/blog/the-keda-autoscaler), which enables workers to auto-scale between 1 and 10 depending on real-time workload. Each worker is a Kubernetes Pod that is hosted within a Kubernetes Node in your Astro Cluster.

Author: ```suggestion
A worker is responsible for executing tasks, which are first scheduled and queued by the Scheduler. On Astro, task execution is powered by the [Celery Executor](https://airflow.apache.org/docs/apache-airflow/stable/executor/celery.html) with [KEDA](https://www.astronomer.io/blog/the-keda-autoscaler), which enables workers to autoscale between 1 and 10 depending on real-time workload. Each worker is a Kubernetes Pod that is hosted within a Kubernetes Node in your Astro Cluster.
```

To modify the resources allocated to the workers in your Deployment, specify a quantity of AUs in the **Worker Resources** field of the Cloud UI. This value determines the size of each worker in your Deployment. While the total number of workers running at any given time may change, all workers will be created with the amount of CPU and memory you specify here. If you set **Worker Resources** to 10 AU, for example, your Deployment might scale up to 3 workers using 10 AU each for a total consumption of 30 AU.

> To ensure reliability, the minimum worker size supported is 10 AU. Beyond that, the maximum worker size you can set depends on the node instance type that is configured for the Cluster in which your Deployment is hosted. If you attempt to provision a worker size that is not supported by your Cluster's instance type, you will see an error in the Cloud UI. For example, if the node instance type for a given Cluster is set to `m5.xlarge`, the maximum worker size supported is 27 AU (2.7 CPUs, 10.1 GiB memory). This limit accounts for some resource overhead that is required for system components.

Author: ```suggestion
To ensure reliability, the minimum worker size supported is 10 AU. Beyond that, the maximum worker size you can set depends on the node instance type that is configured for the Cluster in which your Deployment is hosted. If you attempt to provision a worker size that is not supported by your Cluster's instance type, you will see an error in the Cloud UI. For example, if the node instance type for a given Cluster is set to `m5.xlarge`, the maximum worker size supported for any Deployment within that Cluster is 27 AU (2.7 CPUs, 10.1 GiB memory). This limit accounts for overhead that is required for system components.
```

> For a list of node instance types that are supported on Astro and their corresponding worker size limits, see [AWS Resource Reference](aws-resource-reference.md#deployment-worker-size-limits). To request a different instance type for your Cluster, reach out to [Astronomer Support](https://support.astronomer.io).

Author: ```suggestion
For a list of supported node instance types and their corresponding worker size limits, see [AWS Resource Reference](aws-resource-reference.md#deployment-worker-size-limits). To request a different instance type for your Cluster, reach out to [Astronomer Support](https://support.astronomer.io).
```
